Good Coding Practices/src/js/scripts.js

```diff
@@ -1,41 +1,113 @@
+//api object
+
+const api = (function(){
+
+const temperature =  Math.floor(Math.random()*40);
+
+const getColor = function() {
+switch(true){
+  case temperature < 10 : return "Blue";break;
+  case temperature > 11 && temperature < 20 : return "Green";break;
+  case temperature > 21 && temperature < 30 : return "Yellow";break;
+  case temperature > 30 : return "red";break;
+  default:return "Blue";break;
+}
+};
+
+return{
+  getColor
+}
+
+})()
+
 // piece object
-const piece = (function() {
+const piece = (function(api) {
   let el = null;
+  const color = api.getColor();
   const init = function(el) {
     this.el = el;
+    this.el.style.backgroundColor = color;
+    this.el.style.borderColor = color;
+
   };
   const moveDelta = function(dx, dy) {
     const pos = this.el.getBoundingClientRect();
-    this.el.style.left = `${pos.left + dx}px`;
-    this.el.style.top = `${pos.top + dy}px`;
+    const el = this.el;
+    let x=pos.left,y=pos.top;
+    h = setInterval(function(){
+      if( dx !== 0){
+        if(x == pos.left + dx){
+          clearInterval(h);
+        }
+        x = x + dx/20;
+        el.style.left = `${x}px`;
+      }
+      if( dy !== 0){
+        if(y == pos.top + dy){
+          clearInterval(h);
+        }
+        y = y + dy/20;
+        el.style.top = `${y}px`;
+      }
+    }, 30);
   };
+
+  const backToStartLocation = function(){
+    this.el.style.top = "100px";
+    this.el.style.left = "50%";
+  
+  }
+
+  const randomLocation = function(){
+    this.el.style.top = Math.floor(Math.random()*500) + "px";
+    this.el.style.left = Math.floor(Math.random()*80) + "%";
+  }
+
+  const showHollow = function(isHollow){
+    if(isHollow)
+    this.el.style.backgroundColor = "white";

```

Comment:
1) white is not the same as transparent
2) You have classes for this
3) Don't use JS for things that can be done with CSS